engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -471,18 +471,47 @@ namespace dmGameObject
         hcollection->m_Collection->m_ToBeDeleted = 1;
     }
 
-    void* GetWorld(HCollection hcollection, uint32_t component_index)
+    uint32_t GetComponentTypeIndex(HCollection hcollection, dmhash_t type_hash)
     {
-        if (component_index < MAX_COMPONENT_TYPES)
+        Register* regist = GetRegister(hcollection);
+        for (uint32_t i = 0; i < regist->m_ComponentTypeCount; ++i)
+        {
+            ComponentType* ct = &regist->m_ComponentTypes[i];
+            if (ct->m_NameHash == type_hash)
+            {
+                return i;
+            }
+        }
+        return 0xFFFFFFFF;
+    }
+
+    void* GetWorld(HCollection hcollection, uint32_t component_type_index)
+    {
+        Register* regist = GetRegister(hcollection);
+        if (component_type_index < regist->m_ComponentTypeCount)
         {
-            return hcollection->m_Collection->m_ComponentWorlds[component_index];
+            return hcollection->m_Collection->m_ComponentWorlds[component_type_index];
```

Comment:
A better name imho. Not to be confused with the component indices used in the collection.